swayfmt/src/comments.rs

```diff
@@ -0,0 +1,66 @@
+use std::fmt::Write;
+use sway_ast::token::Comment;
+use sway_types::Spanned;
+
+use crate::{formatter::FormattedCode, Formatter, FormatterError};
+
+// Given a start, an end and a CommentMap, return references to all comments
+// contained within a given start and end of a span, in an exclusive range.
+pub fn get_comments_between(start: usize, end: usize, formatter: &mut Formatter) -> Vec<&Comment> {
+    let mut comments = vec![];
+    let iter = formatter.comment_map.clone().into_keys();
+
+    for bs in iter {
+        if bs.start > start && bs.end < end {
+            if let Some(comment) = formatter.comment_map.get(&bs) {
+                comments.push(comment)
+            }
+        }
+    }
+
+    comments
+}
+
+// Given a start, an end and a CommentMap, removes and returns all comments in a
+// CommentMap contained within a given start and end of a span, in an exclusive range.
+pub fn take_comments_between(start: usize, end: usize, formatter: &mut Formatter) -> Vec<Comment> {
+    let mut comments = vec![];
+    let iter = formatter.comment_map.clone().into_keys();
+
+    for bs in iter {
+        if bs.start > start && bs.end < end {
+            if let Some(comment) = formatter.comment_map.remove(&bs) {
+                comments.push(comment)
+            }
+        }
+    }
+
+    comments
+}
+
+// Writes comments between a given start and end and removes them from the formatter's CommentMap.
+// This returns Ok(true) on successful execution AND comments were written, but returns
+// Ok(false) on successful execution without any comments written. Returns Err on failure.
+pub fn maybe_write_comments_from_map(
+    formatted_code: &mut FormattedCode,
+    start: usize,
+    end: usize,
```

Comment:
Rather than specifying `start` and `end` separately, another option is to accept a range, e.g.

```suggestion
    range: std::ops::Range<usize>,
```